backend/learnings/views.py

```diff
@@ -178,10 +178,25 @@ def get(self, request):
         return Response(serializer.data)
 
 
-class UserDetails(generics.RetrieveAPIView):
-    queryset = EUser.objects.all()
-    serializer_class = EUserSerializer
-    lookup_field = "username"
+class UserDetails(APIView):
+    permission_classes = [IsAuthenticated]
+    authentication_classes = [TokenAuthentication]
+
+    def get(self, request, username, follower_username, following_username):
+        user = EUser.objects.get(username=username)
+        serializer = EUserSerializer(user)
+        if follower_username != "none" or following_username != "none":
+            try:
+                check = Follow.objects.get(
+                    follower_id__username=follower_username,
+                    following_id__username=following_username,
+                )
+                content = {"status": "unfollow", "data": serializer.data}
+                return Response(content)
+            except Exception:
+                content = {"status": "follow", "data": serializer.data}
+                return Response(content)
```

Comment:
pa move outside of try catch po ung return, let me know if it raised an error po :bow: 
```
try:
    do domething
    content = something
catch:
    do something
    content= something
return Resposne(content)
```